caching-service/src/test/java/org/zowe/apiml/caching/service/infinispan/storage/InfinispanStorageTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.caching.service.infinispan.storage;
+
+import org.infinispan.Cache;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.zowe.apiml.caching.model.KeyValue;
+import org.zowe.apiml.caching.service.StorageException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class InfinispanStorageTest {
+
+    public static final KeyValue TO_CREATE = new KeyValue("key1", "val1");
+    public static final KeyValue TO_UPDATE = new KeyValue("key1", "val2");
+    Cache<String, Map<String, KeyValue>> cache;
+    InfinispanStorage storage;
+    String serviceId1 = "service1";
+
+    @BeforeEach
+    void setup() {
+        cache = mock(Cache.class);
+        storage = new InfinispanStorage(cache);
+    }
+
+    @Nested
+    class WhenEntryDoesntExist {
+
+        Map<String, KeyValue> serviceStore;
+
+        @BeforeEach
+        void createEmptyStore() {
+            serviceStore = new HashMap<>();
+        }
+
+        @Test
+        void whenCreate_thenCacheIsUpdated() {
```

Comment:
Just a suggestion on the method naming. Here and in the methods below is redundant as the `when` word is already used in the class name. Consider creating maybe nested classes for create, read etc.

Reply:
Test groups are based on cache state(exists/doesn't exist) and not on the methods. I wouldn't say that "when" is redundant here, "when" in class name is referring to cache initial state, but "when" in the method is referring to the action executed. Creating groups per method would result in more groups than tests, and I think this is not better result.

Reply:
yes I understand, In this case is also hard to follow the `given_when_then`.  I'm ok with this then.